vault/resource_aws_secret_backend.go

```diff
@@ -169,36 +181,17 @@ func awsSecretBackendCreate(ctx context.Context, d *schema.ResourceData, meta in
 	}
 
 	path := d.Get(consts.FieldPath).(string)
-	description := d.Get(consts.FieldDescription).(string)
-	defaultTTL := d.Get(consts.FieldDefaultLeaseTTL).(int)
-	maxTTL := d.Get(consts.FieldMaxLeaseTTL).(int)
 	accessKey := d.Get(consts.FieldAccessKey).(string)
 	secretKey := d.Get(consts.FieldSecretKey).(string)
 	region := d.Get(consts.FieldRegion).(string)
-	local := d.Get(consts.FieldLocal).(bool)
 
 	d.Partial(true)
 	log.Printf("[DEBUG] Mounting AWS backend at %q", path)
-	mountConfig := api.MountConfigInput{
-		DefaultLeaseTTL: fmt.Sprintf("%ds", defaultTTL),
-		MaxLeaseTTL:     fmt.Sprintf("%ds", maxTTL),
-	}
-	useAPIVer116 := provider.IsAPISupported(meta, provider.VaultVersion116) && provider.IsEnterpriseSupported(meta)
-	if useAPIVer116 {
-		identityTokenKey := d.Get(consts.FieldIdentityTokenKey).(string)
-		if identityTokenKey != "" {
-			mountConfig.IdentityTokenKey = identityTokenKey
-		}
-	}
-	err := client.Sys().MountWithContext(ctx, path, &api.MountInput{
-		Type:        consts.MountTypeAWS,
-		Description: description,
-		Local:       local,
-		Config:      mountConfig,
-	})
-	if err != nil {
-		return diag.Errorf("error mounting to %q: %s", path, err)
+
+	if err := createMount(d, meta, client, path, consts.MountTypeAWS); err != nil {
```

Comment:
Looks like we are going from using `MountWithContext` to the context-less mount. Should we update `createMount()` to also take a context.

Reply:
Updated all create instances to also use the context